third_party/move/tools/move-smith/src/ast.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{names::Identifier, types::Type};
+use arbitrary::Arbitrary;
+use num_bigint::BigUint;
+
+#[derive(Debug, Clone)]
+pub struct CompileUnit {
```

Comment:
All public structs, functions, methods, enums, etc should have at least a brief documentation. They should list any invariants to be maintained on datastructures, pre-conditions and post-conditions if applicable.

Reply:
Added